### ISSUE TITLE
Performance issues when starting a station with queue open

### ISSUE THREAD

**Issue by ronilaukkarinen** [radiant-player/radiant-player-mac#625]
Almost every time when I start a station and queue opens, Radiant gets unresponsive and hogs up memory. Many times I get the distorted non-musical "unresponsiveness" sound when this happens, and Dock is showing app "Not responding". Then when it wakes up and music continues to play, I close the queue and everything works smoothly again.

I noticed Last.fm have been having problems today and yesterday, could it be causing this? (real time scrobbling) just a hunch, because my python script for Pandora does similar things when Last.fm has problems.

**System**: OS X 10.11.6 (El Capitan)
**App**: 1.11.2

**Comment by ronilaukkarinen** [comment]
Confirmed that seeing the "rainbow of death" every once in a while even though queue not open. Also the annoying "out of memory" sound sometimes in the middle of the song, randomly. How this could be improved? a bit more memory optimization?

I have quite massive library though.

**Comment by jacobwgillespie** [comment]
Can you try this build of the very early Radiant v2 to see if you experience the same performance issues?  https://dl.dropboxusercontent.com/u/6278910/Radiant%20Player-2.0.0-alpha-2.dmg

**Comment by ronilaukkarinen** [comment]
Tested and seems to work fine. Perhaps too alpha though, can't add theme or Last.fm? curious to see how it performs with other features that may be the culprit here. Building everything from scratch or what's the intention here?

**Comment by jacobwgillespie** [comment]
Awesome!  Yeah, I'm not expecting performance to change much as the additional features like themes and Last.fm are very resource-light.

It is a complete rewrite, yes, and the main reason is that Google makes GPM essentially for Chrome.  It's optimized for Chrome, and other browsers often have weird issues.  Radiant v1 uses embedded Safari to display the player, and that causes a bunch of issues that are often out of our control to fix, including a lot of the performance issues.  Radiant v2 is a rewrite that utilizes the Chrome runtime under the hood (well, Chromium) and thus takes advantage of "native" support from Google.  I need to reimplement the features now, like the settings dialog and Last.fm integration.  Just have to find the time.  :)

**Comment by ronilaukkarinen** [comment]
Oh, ok. I am mainly using this because of the different approach to [Google-Play-Music-Desktop-Player-UNOFFICIAL](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-) which is built on Electron as well. It has mostly same things than Radiant, but also some amount of bugs, that's why I've been using Radiant Player - less buggy ;)

**Comment by jacobwgillespie** [comment]
Yep, once we make the v2 switch we'll be closer Google-Play...UNOFFICIAL as far as the underlying browser engine goes.  Still planning to keep the same features / look-and-feel from Radiant v1, though, so it will essentially be a faster / more stable Radiant.

**Comment by MarshallOfSound** [comment]
> It has mostly same things than Radiant, but also some amount of bugs

Can't fix 'em if you don't raise 'em 👍 😆
